crates/ruma-macros/src/api/request.rs

```diff
@@ -62,28 +61,28 @@ pub fn expand_derive_request(input: DeriveInput) -> syn::Result<TokenStream> {
             continue;
         }
 
-        let meta = attr.parse_args_with(Punctuated::<_, Token![,]>::parse_terminated)?;
-        for MetaNameValue { name, value } in meta {
-            match value {
-                MetaValue::Type(t) if name == "authentication" => {
+        let metas =
+            attr.parse_args_with(Punctuated::<DeriveRequestMeta, Token![,]>::parse_terminated)?;
+        for meta in metas {
+            match meta {
+                DeriveRequestMeta::Authentication(t) => {
                     authentication = Some(parse_quote!(#t));
```

Comment:
I wonder whether we should remove these semicolons to get more compact formatting 🤔